bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsBinarySyncImpl.java

```diff
@@ -0,0 +1,209 @@
+)
+@Service
+public class RemoteAssetsBinarySyncImpl implements RemoteAssetsBinarySync {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteAssetsBinarySyncImpl.class);
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Reference
+    private RemoteAssetsConfig remoteAssetsConfig;
+
+    private ResourceResolver resourceResolver;
+    private Session session;
+
+    /**
+     * Method to run on activation.
+     * @throws RepositoryException exception
+     */
+    @Activate
+    protected void activate() throws RepositoryException {
+        this.resourceResolver = RemoteAssets.logIn(this.resourceResolverFactory);
+        this.session = this.resourceResolver.adaptTo(Session.class);
+        if (StringUtils.isNotBlank(this.remoteAssetsConfig.getEventUserData())) {
+            this.session.getWorkspace().getObservationManager().setUserData(this.remoteAssetsConfig.getEventUserData());
+        }
+    }
+
+    /**
+     * Method to run on deactivation.
+     */
+    @Deactivate
+    protected void deactivate() {
+        if (this.session != null) {
+            try {
+                this.session.logout();
+            } catch (Exception e) {
+                LOG.warn("Failed session.logout()", e);
+            }
+        }
+        if (this.resourceResolver != null) {
+            try {
+                this.resourceResolver.close();
+            } catch (Exception e) {
+                LOG.warn("Failed resourceResolver.close()", e);
+            }
+        }
+    }
+
+    /**
+     * @see RemoteAssetsBinarySync#syncAsset(Resource)
+     * @param resource Resource
+     * @return Resource
+     */
+    @Override
+    public Resource syncAsset(Resource resource) {
+        try {
+            this.session.refresh(true);
+            Resource localRes = this.resourceResolver.getResource(resource.getPath());
+            Node node = localRes.adaptTo(Node.class);
+
+            Asset asset = localRes.getParent().adaptTo(Asset.class);
+            URI pathUi = new URI(null, null, asset.getPath(), null);
+            String baseUrl = this.remoteAssetsConfig.getServer().concat(pathUi.toString()).concat("/_jcr_content/renditions/");
+
+            Iterator<? extends Rendition> renditions = asset.listRenditions();
+            while (renditions.hasNext()) {
+                Rendition assetRendition = renditions.next();
+                if (StringUtils.isEmpty(assetRendition.getMimeType())) {
+                    continue;
+                }
+                String renditionName = assetRendition.getName();
+
+                URL url = new URL(String.format("%s%s", baseUrl, renditionName));
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod("GET");
+                connection.setRequestProperty("Authorization", String.format("Basic %s", RemoteAssets.encodeForBasicAuth(this.remoteAssetsConfig)));
```

Comment:
Feels like it should WARN or something if the connection is NOT https... since the password would be compromisable.

Reply:
we can do that - should we also make it so that HTTPS is required unless a user configures specifically to allow HTTP?  For internal network traffic within a VPN it might not be a problem to be HTTP.

Or do you have alternate suggestions on how we authenticate?